### DISCUSSION TITLE
Invalid `prisma.course.findUnique()` invocation while trying to fetch

### DISCUSSION THREAD

**Discussion by Akhil7K** [prisma/prisma#22325] [Q&A]
### Question

While trying to find the unique course which is an unique field defined in the prisma schema file.
`id` is defined as INT in course model in prisma schema file.
`courseId` has been defined as number in page.tsx (error arises here).

Error in terminal: 
```js
`Invalid `prisma.course.findUnique()` invocation:

{
  where: {
    id: "1",
        ~~~
    userId: "user_2YTcfLAelOkdGjy6SAaUpztHPxc"
  },
  include: {
    chapters: {
      where: {
        isPublished: true
      },
      orderBy: {
        position: "asc"
      }
    },
    attachments: {
      orderBy: {
        createdAt: "desc"
      }
    }
  }
}

Argument `id`: Invalid value provided. Expected Int, provided String.`
```
As it can be seen in the above provided error message `id` is supposed to be a number but a string is being returned which I believe is the cause for the error. I've been through my schema.prisma file and other related files where I have used courseId as a param. page.tsx is the sole file throwing this error. 

### Expected behavior (optional)

Course creation page is supposed to load without any errors.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```
![Screenshot 2023-12-09 112540](https://github.com/prisma/prisma/assets/146330175/e944a43e-6254-49d2-8f62-5983afd1eb3c)

```ts
// Add any relevant Prisma Client queries here
```

This is the concerned course model, as it can be seen id has been defined as an INT type, and the subsequent relation betweeen models also maintains the same INT type.
![Screenshot 2023-12-09 112557](https://github.com/prisma/prisma/assets/146330175/f4100763-27dd-464d-bfd8-511599d6ce2b)

- OS: Windows 11
- Database: MySQL
- Node.js version: v20.7.0

- Run `prisma -v` to see your Prisma version and paste it
- prisma                  : 5.7.0
@prisma/client          : 5.7.0
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.7.0
Query Engine (Node-API) : libquery-engine 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.7.0-41.79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Default Engines Hash    : 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Studio                  : 0.495.0
Preview Features        : fullTextIndex, fullTextSearch

**Comment by shulinnn** [comment]
Didnt noticed this the first time but if u input "1" its not the same as 1 , so if u change `id : 1 `instead of `id : "1"` should work.

**Reply by Akhil7K** [reply]
the courseId is being fetched from my db where I've defined it as a Int and in my page.tsx I've defined the type of courseId as number but still it returns a string, and I don't get it why?

**Reply by ludralph** [reply]
HI @Akhil7K 👋 

Looking at the error, it appears the string "1" is being passed to the query. Can you please share the code you are using to perform the query? You may need to cast the value being passed to the `id` field in the `where` clause.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
